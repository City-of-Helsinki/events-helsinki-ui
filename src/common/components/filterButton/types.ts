export type FilterType =
  | 'category'
  | 'date'
  | 'dateType'
  | 'division'
  | 'place'
  | 'publisher'
  | 'target'
  | 'text'
  | 'minAge'
  | 'maxAge'
  | 'exactAge';
