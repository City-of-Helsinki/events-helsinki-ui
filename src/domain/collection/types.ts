import { CollectionDetails, LocalizedObject } from "../../generated/graphql";

export type CollectionInList = Pick<
  CollectionDetails,
  "id" | "curatedEvents" | "eventListQuery"
> & { title: Pick<LocalizedObject, "en" | "fi" | "sv"> };
