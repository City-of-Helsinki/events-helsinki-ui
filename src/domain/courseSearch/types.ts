export interface Filters {
  categories: string[];
  dateTypes: string[];
  divisions: string[];
  places: string[];
  text: string[];
  start: Date | null;
  end: Date | null;
  isFree?: boolean;
  alsoOngoingCourses?: boolean;
}

export interface MappedFilters {
  categories: string[];
  dateTypes?: string[];
  divisions: string[];
  isFree?: boolean;
  alsoOngoingCourses?: boolean;
  places: string[];
  publisher?: string | null;
  text: string[];
  start: string | null;
  end: string | null;
}
