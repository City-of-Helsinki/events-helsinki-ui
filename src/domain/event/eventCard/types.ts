import { ROUTES } from '../../app/constants';

export enum EventType {
  EVENT = 'event',
  COURSE = 'course',
}

export const EVENT_ROUTE_MAPPER = {
  [EventType.EVENT]: ROUTES.EVENT,
  [EventType.COURSE]: ROUTES.COURSE,
};

export const EVENTS_ROUTE_MAPPER = {
  [EventType.EVENT]: ROUTES.EVENTS,
  [EventType.COURSE]: ROUTES.COURSES,
};
