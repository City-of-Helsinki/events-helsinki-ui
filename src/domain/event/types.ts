import {
  Division,
  EventDetails,
  Image,
  Keyword,
  LocalizedObject,
  Maybe,
  Offer
} from "../../generated/graphql";

type OfferInList = Pick<Offer, "isFree"> & {
  description: Maybe<LocalizedObject>;
  price: Maybe<LocalizedObject>;
  infoUrl: Maybe<LocalizedObject>;
};

type KeywordInList = Pick<Keyword, "id"> & {
  name: Maybe<LocalizedObject>;
};

type LocationInList = {
  addressLocality: Maybe<LocalizedObject>;
  divisions: Maybe<
    Array<
      Pick<Division, "type"> & {
        name: Maybe<LocalizedObject>;
      }
    >
  >;
  name: Maybe<LocalizedObject>;
  streetAddress: Maybe<LocalizedObject>;
};

export type EventUiKeyword = {
  id: string;
  name: string;
};

export type EventInList = Pick<EventDetails, "id" | "startTime" | "endTime"> & {
  images: Array<Pick<Image, "id" | "name" | "url">>;
  keywords: Array<KeywordInList>;
  location: Maybe<LocationInList>;
  name: LocalizedObject;
  offers: Array<OfferInList>;
};
