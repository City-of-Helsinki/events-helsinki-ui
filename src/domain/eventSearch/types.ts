import {
  COURSE_CATEGORIES,
  COURSE_HOBBY_TYPES,
  EVENT_CATEGORIES,
} from './constants';

export interface CategoryOption {
  icon: React.ReactElement;
  text: string;
  value: EVENT_CATEGORIES | COURSE_CATEGORIES;
}

export interface HobbyTypeOption {
  icon: React.ReactElement;
  text: string;
  value: COURSE_HOBBY_TYPES;
}

export interface Filters {
  categories: string[];
  hobbyTypes?: string[];
  dateTypes: string[];
  divisions: string[];
  end: Date | null;
  isFree?: boolean;
  keyword?: string[];
  keywordNot?: string[];
  onlyChildrenEvents?: boolean;
  onlyEveningEvents?: boolean;
  alsoOngoingCourses?: boolean;
  places: string[];
  publisher?: string | null;
  start: Date | null;
  text: string[];
  minAge?: string;
  maxAge?: string;
}

export interface MappedFilters {
  categories: string[];
  hobbyTypes?: string[];
  dateTypes?: string[];
  divisions: string[];
  end?: string | null;
  isFree?: boolean;
  keyword?: string[];
  keywordNot?: string[];
  onlyChildrenEvents?: boolean;
  onlyEveningEvents?: boolean;
  alsoOngoingCourses?: boolean;
  places: string[];
  publisher?: string | null;
  start?: string | null;
  text: string[];
}
