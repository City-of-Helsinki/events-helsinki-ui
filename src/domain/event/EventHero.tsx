import classNames from "classnames";
import { isThisWeek, isToday } from "date-fns";
import {
  IconAngleRight,
  IconArrowLeft,
  IconLocation,
  IconTicket
} from "hds-react";
import React from "react";
import { useTranslation } from "react-i18next";
import { RouteComponentProps, withRouter } from "react-router";

import Button from "../../common/components/button/Button";
import { EventDetailsQuery } from "../../generated/graphql";
import getDateRangeStr from "../../util/getDateRangeStr";
import getLocale from "../../util/getLocale";
import getLocalisedString from "../../util/getLocalisedString";
import Container from "../app/layout/Container";
import styles from "./eventHero.module.scss";
import EventKeywords from "./EventKeywords";
import LocationText from "./EventLocationText";
import { getEventPrice } from "./EventUtils";

interface Props extends RouteComponentProps {
  eventData: EventDetailsQuery;
}

const EventHero: React.FC<Props> = ({
  eventData,
  history: { push },
  location: { search }
}) => {
  const { t } = useTranslation();
  const locale = getLocale();

  const offerInfoUrl = React.useMemo(() => {
    const offer = eventData.eventDetails.offers.find(item =>
      getLocalisedString(item.infoUrl || {}, locale)
    );

    return offer ? getLocalisedString(offer.infoUrl || {}, locale) : "";
  }, [eventData.eventDetails.offers, locale]);

  const handleBack = () => {
    push({ pathname: `/${locale}/events`, search });
  };

  const moveToBuyTicketsPage = () => {
    window.open(offerInfoUrl);
  };

  const image = eventData.eventDetails.images.length
    ? eventData.eventDetails.images[0]
    : null;
  const description = eventData.eventDetails.shortDescription || {};
  const keywords = eventData.eventDetails.keywords;
  const startTime = eventData.eventDetails.startTime;
  const endTime = eventData.eventDetails.endTime;
  const name = eventData.eventDetails.name;
  const today = startTime ? isToday(new Date(startTime)) : false;
  const thisWeek = startTime ? isThisWeek(new Date(startTime)) : false;

  return (
    <div className={styles.heroWrapper}>
      <Container>
        <div className={styles.contentWrapper}>
          <div className={styles.backButtonWrapper}>
            <button className={styles.backButton} onClick={handleBack}>
              <IconArrowLeft />
            </button>
          </div>
          <div>
            {image && (
              <div
                className={styles.image}
                style={{ backgroundImage: `url(${image.url})` }}
              />
            )}
          </div>
          <div className={styles.leftPanel}>
            <div className={styles.textWrapper}>
              {(today || thisWeek || (!!keywords && !!keywords.length)) && (
                <div className={styles.categoryWrapper}>
                  <EventKeywords
                    blackOnMobile={true}
                    event={eventData.eventDetails}
                    showIsFree={false}
                  />
                </div>
              )}
              <div className={classNames(styles.date, styles.desktopOnly)}>
                {!!startTime && getDateRangeStr(startTime, endTime, locale)}
              </div>
              <div className={styles.title}>
                {getLocalisedString(name, locale)}
              </div>
              <div className={styles.description}>
                {getLocalisedString(description, locale)}
              </div>

              <div
                className={classNames(styles.infoWithIcon, styles.desktopOnly)}
              >
                <div className={styles.iconWrapper}>
                  <IconLocation className={styles.icon} />
                </div>
                <div className={styles.iconTextWrapper}>
                  <LocationText
                    event={eventData.eventDetails}
                    showDistrict={false}
                    showLocationName={true}
                  />
                </div>
              </div>

              <div
                className={classNames(styles.infoWithIcon, styles.desktopOnly)}
              >
                <div className={styles.iconWrapper}>
                  <IconTicket className={styles.icon} />
                </div>
                <div className={styles.iconTextWrapper}>
                  {getEventPrice(
                    eventData.eventDetails,
                    locale,
                    t("event.hero.offers.isFree")
                  ) || "-"}
                </div>
              </div>
              {!!offerInfoUrl && (
                <>
                  <div
                    className={classNames(
                      styles.buyButtonWrapper,
                      styles.desktopOnly
                    )}
                  >
                    <Button
                      color="primary"
                      iconRight={<IconAngleRight />}
                      onClick={moveToBuyTicketsPage}
                      size="default"
                    >
                      {t("event.hero.buttonBuyTickets")}
                    </Button>
                  </div>
                </>
              )}
            </div>
          </div>
        </div>
      </Container>
    </div>
  );
};

export default withRouter(EventHero);
