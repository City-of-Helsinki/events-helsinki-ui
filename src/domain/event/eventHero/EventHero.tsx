import { Button, IconArrowLeft, IconLocation, IconTicket } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory, useLocation } from 'react-router-dom';

import IconButton from '../../../common/components/iconButton/IconButton';
import InfoWithIcon from '../../../common/components/infoWithIcon/InfoWithIcon';
import Visible from '../../../common/components/visible/Visible';
import { EventFieldsFragment } from '../../../generated/graphql';
import useLocale from '../../../hooks/useLocale';
import getDateRangeStr from '../../../util/getDateRangeStr';
import testImage from '../../../util/testImage';
import Container from '../../app/layout/Container';
import EventKeywords from '../eventKeywords/EventKeywords';
import LocationText from '../eventLocation/EventLocationText';
import EventName from '../eventName/EventName';
import {
  extractLatestReturnPath,
  ReturnParams,
} from '../eventQueryString.util';
import { getEventFields, getEventPrice } from '../EventUtils';
import styles from './eventHero.module.scss';

interface Props {
  event: EventFieldsFragment;
}

const EventHero: React.FC<Props> = ({ event }) => {
  const { t } = useTranslation();
  const [showBackupImage, setShowBackupImage] = React.useState(false);
  const locale = useLocale();
  const history = useHistory();
  const { search } = useLocation();

  const {
    endTime,
    imageUrl,
    keywords,
    offerInfoUrl,
    placeholderImage,
    shortDescription,
    startTime,
    today,
    thisWeek,
    showBuyButton,
  } = getEventFields(event, locale);
  const eventPriceText = getEventPrice(
    event,
    locale,
    t('event.hero.offers.isFree')
  );
  const showKeywords = Boolean(today || thisWeek || keywords.length);

  const returnParam = extractLatestReturnPath(search);

  const goBack = ({ returnPath, remainingQueryString }: ReturnParams) => {
    history.push({
      pathname: `/${locale}${returnPath}`,
      search: remainingQueryString,
      state: { eventId: event.id },
    });
  };

  const goToBuyTicketsPage = () => {
    window.open(offerInfoUrl);
  };

  React.useEffect(() => {
    if (imageUrl) {
      const testThatImageExist = async () => {
        try {
          await testImage(imageUrl);
        } catch {
          setShowBackupImage(true);
        }
      };

      testThatImageExist();
    }
  }, [imageUrl]);

  return (
    <div className={styles.heroWrapper}>
      <Container>
        <div className={styles.contentWrapper}>
          <div className={styles.backButtonWrapper}>
            <IconButton
              ariaLabel={t('event.hero.ariaLabelBackButton')}
              backgroundColor="white"
              icon={<IconArrowLeft />}
              onClick={() => goBack(returnParam)}
              size="default"
            />
          </div>
          <div>
            <div
              className={styles.image}
              style={{
                backgroundImage: `url(${
                  showBackupImage ? placeholderImage : imageUrl
                })`,
              }}
            />
          </div>
          <div className={styles.leftPanel}>
            <div className={styles.textWrapper}>
              <h1 className={styles.title}>
                <EventName event={event} />
              </h1>
              <div className={styles.description}>{shortDescription}</div>
              <Visible above="sm" className={styles.date}>
                {!!startTime &&
                  getDateRangeStr({
                    start: startTime,
                    end: endTime,
                    locale,
                    includeTime: true,
                    timeAbbreviation: t('commons.timeAbbreviation'),
                  })}
              </Visible>

              <Visible above="sm" className={styles.location}>
                <InfoWithIcon icon={<IconLocation />} title={''}>
                  <LocationText
                    event={event}
                    showDistrict={false}
                    showLocationName={true}
                  />
                </InfoWithIcon>
              </Visible>

              <Visible above="sm" className={styles.price}>
                <InfoWithIcon icon={<IconTicket />} title={''}>
                  {eventPriceText || '-'}
                </InfoWithIcon>
              </Visible>

              {showKeywords && (
                <div className={styles.categoryWrapper}>
                  <EventKeywords
                    blackOnMobile={true}
                    event={event}
                    showIsFree={true}
                  />
                </div>
              )}
              {showBuyButton && (
                <Visible above="sm" className={styles.buyButtonWrapper}>
                  <Button
                    aria-label={t('event.hero.ariaLabelBuyTickets')}
                    onClick={goToBuyTicketsPage}
                    variant="success"
                  >
                    {t('event.hero.buttonBuyTickets')}
                  </Button>
                </Visible>
              )}
            </div>
          </div>
        </div>
      </Container>
    </div>
  );
};

export default EventHero;
