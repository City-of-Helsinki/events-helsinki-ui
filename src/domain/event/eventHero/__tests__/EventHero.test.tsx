import { advanceTo, clear } from 'jest-date-mock';
import capitalize from 'lodash/capitalize';
import * as React from 'react';

import translations from '../../../../common/translation/i18n/fi.json';
import {
  EventFieldsFragment,
  OfferFieldsFragment,
} from '../../../../generated/graphql';
import {
  fakeEvent,
  fakeKeyword,
  fakeOffer,
} from '../../../../util/mockDataUtils';
import { render, screen, userEvent } from '../../../../util/testUtils';
import { EventType } from '../../types';
import EventHero, { Props as EventHeroProps } from '../EventHero';

const name = 'Event name';
const startTime = '2020-06-22T07:00:00.000000Z';
const endTime = '2020-06-22T10:00:00.000000Z';
const shortDescription = 'Event description';
const locationName = 'Location name';
const streetAddress = 'Test address 1';
const addressLocality = 'Helsinki';

const keywordNames = ['keyword 1', 'keyword 2'];
const keywords = keywordNames.map((name) =>
  fakeKeyword({ name: { fi: name } })
);

const event = fakeEvent({
  name: { fi: name },
  keywords,
  startTime,
  endTime,
  publisher: '',
  shortDescription: { fi: shortDescription },
  location: {
    internalId: 'tprek:8740',
    addressLocality: { fi: addressLocality },
    name: { fi: locationName },
    streetAddress: { fi: streetAddress },
  },
}) as EventFieldsFragment;

afterAll(() => {
  clear();
});

const renderComponent = (props?: Partial<EventHeroProps>) => {
  return render(
    <EventHero event={event} eventType={EventType.EVENT} {...props} />
  );
};

test('should render event name, description and location', () => {
  renderComponent();

  expect(screen.queryByRole('heading', { name })).toBeInTheDocument();
  expect(screen.queryByText(shortDescription)).toBeInTheDocument();
  expect(
    screen.queryByText(
      [locationName, streetAddress, addressLocality].join(', ')
    )
  ).toBeInTheDocument();
});

test('should go to event list', () => {
  const { history } = renderComponent();

  userEvent.click(
    screen.getByRole('button', {
      name: translations.event.hero.ariaLabelBackButton,
    })
  );
  expect(history.location.pathname).toBe('/fi/events');
});

test('should render keywords', () => {
  renderComponent();

  keywordNames.forEach((keyword) => {
    expect(screen.queryByText(capitalize(keyword))).toBeInTheDocument();
  });
});

test('should render today tag', () => {
  advanceTo('2020-06-22');
  renderComponent();

  expect(
    screen.queryByRole('button', {
      name: translations.event.categories.labelToday,
    })
  ).toBeInTheDocument();
  expect(
    screen.queryByRole('button', {
      name: translations.event.categories.labelThisWeek,
    })
  ).not.toBeInTheDocument();
});

test('should render this week tag', () => {
  advanceTo('2020-06-23');
  renderComponent();

  expect(
    screen.queryByRole('button', {
      name: translations.event.categories.labelToday,
    })
  ).not.toBeInTheDocument();
  expect(
    screen.queryByRole('button', {
      name: translations.event.categories.labelThisWeek,
    })
  ).toBeInTheDocument();
});

test('should hide buy button for free events', () => {
  const mockEvent = {
    ...event,
    offers: [fakeOffer({ isFree: true }) as OfferFieldsFragment],
  };
  render(<EventHero event={mockEvent} eventType="event" />);

  expect(
    screen.queryByRole('button', {
      name: new RegExp(translations.event.hero.buttonBuyTickets, 'i'),
    })
  ).not.toBeInTheDocument();
});

test('should show buy button', () => {
  global.open = jest.fn();
  const mockEvent = {
    ...event,
    offers: [fakeOffer({ isFree: false }) as OfferFieldsFragment],
  };

  render(<EventHero event={mockEvent} eventType="event" />);

  userEvent.click(
    screen.queryByRole('button', {
      name: new RegExp(translations.event.hero.buttonBuyTickets, 'i'),
    })
  );
  expect(global.open).toBeCalledTimes(1);
});
