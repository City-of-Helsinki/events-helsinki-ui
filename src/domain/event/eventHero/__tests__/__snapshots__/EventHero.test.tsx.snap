// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventHero matches snapshot 1`] = `
<div
  className="heroWrapper"
>
  <div
    className="container"
  >
    <div
      className="contentWrapper"
    >
      <div
        className="backButtonWrapper"
      >
        <a
          aria-label="Takaisin tapahtumahakuun"
          className="iconLink white"
          href="/fi/events"
          onClick={[Function]}
        >
          <div
            className="iconWrapper"
          >
            <svg
              className=""
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <polygon
                  points="13.3706831 2 11.5237192 3.84696395 18.0000474 10.3242884 2 10.3242884 2 12.937334 17.9990512 12.937334 11.5237192 19.412666 13.3706831 21.2586338 23 11.6293169"
                  transform="translate(12.500000, 11.629317) rotate(-180.000000) translate(-12.500000, -11.629317) "
                />
              </g>
            </svg>
          </div>
        </a>
      </div>
      <div>
        <div
          className="image"
          style={
            Object {
              "backgroundImage": "url(https://api.hel.fi/linkedevents/media/images/piha-2_yYlCdPB.jpg)",
            }
          }
        />
      </div>
      <div
        className="leftPanel"
      >
        <div
          className="textWrapper"
        >
          <div
            className="categoryWrapper"
          >
            <div
              className="keyword blackOnMobile"
            >
              Keyword 1 fi
            </div>
            <div
              className="keyword blackOnMobile"
            >
              Keyword 2 fi
            </div>
            <div
              className="keyword tramLight20"
            >
              Ilmainen
            </div>
          </div>
          <div
            className="date desktopOnly"
          >
            Pe 13.12.2019
          </div>
          <div
            className="title"
          >
            name fi
          </div>
          <div
            className="description"
          >
            short description fi
          </div>
          <div
            className="infoWithIcon desktopOnly"
          >
            <div
              className="iconWrapper"
            >
              <svg
                className="icon"
                height="100%"
                viewBox="0 0 23 23"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.6911445,6.57252031e-13 C13.9380141,6.57252031e-13 16.1856662,0.848347826 17.8995793,2.54582609 C21.3274054,5.94 21.0957532,11.2296522 17.8995793,14.8382609 C14.618101,18.5446957 12.868188,20.8565217 11.6911445,23.1793043 C10.5203619,20.8565217 8.7657532,18.5446957 5.48505755,14.8382609 C2.28888364,11.2296522 2.05644886,5.94 5.48505755,2.54582609 C7.19897059,0.848347826 9.44584016,6.57252031e-13 11.6911445,6.57252031e-13 Z M11.6911445,1.73895652 C9.80818799,1.73895652 8.03792712,2.46443478 6.70827494,3.78078261 C5.39505755,5.08069565 4.71418799,6.74295652 4.73923146,8.58913043 C4.76349233,10.4126087 5.49053581,12.222 6.78653581,13.6862609 C9.03966625,16.2305217 10.568101,18.1166087 11.6927097,19.7773043 C12.8220141,18.111913 14.348101,16.2266087 16.598101,13.6862609 C17.894101,12.222 18.6211445,10.4126087 18.6454054,8.58913043 C18.6688836,6.74295652 17.9880141,5.08069565 16.6763619,3.78078261 C15.3451445,2.46443478 13.5756662,1.73895652 11.6911445,1.73895652 Z M11.6922402,5.21538261 C13.6127619,5.21538261 15.168588,6.7719913 15.168588,8.69251304 C15.168588,10.6122522 13.6127619,12.1688609 11.6922402,12.1688609 C9.77171842,12.1688609 8.21510973,10.6122522 8.21510973,8.69251304 C8.21510973,6.7719913 9.77171842,5.21538261 11.6922402,5.21538261 Z M11.6922402,6.95355652 C10.7327619,6.95355652 9.95328364,7.73381739 9.95328364,8.69251304 C9.95328364,9.65042609 10.7327619,10.430687 11.6922402,10.430687 C12.6509358,10.430687 13.4304141,9.65042609 13.4304141,8.69251304 C13.4304141,7.73381739 12.6509358,6.95355652 11.6922402,6.95355652 Z"
                />
              </svg>
            </div>
            <div
              className="iconTextWrapper"
            >
              name fi, streetAddress fi, locality fi
            </div>
          </div>
          <div
            className="infoWithIcon desktopOnly"
          >
            <div
              className="iconWrapper"
            >
              <svg
                className="icon"
                height="100%"
                viewBox="0 0 24 24"
                width="100%"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fillRule="evenodd"
                  strokeWidth="1"
                >
                  <path
                    d="M15.347542,1 L18.2345112,3.88934482 L17.745134,4.37990983 C17.7418676,4.38317631 17.7383041,4.38673973 17.7350377,4.39000621 C17.2257647,4.90967254 17.2287342,5.74707772 17.7448371,6.26318063 C17.9954647,6.51440218 18.3298329,6.6527819 18.6858786,6.6527819 C19.0422212,6.65307885 19.3765894,6.51499609 19.627514,6.26377453 L19.627514,6.26377453 L19.844586,6.04759334 C19.8831898,6.00988041 19.9220905,5.97038577 19.9333747,5.95999244 L19.9333747,5.95999244 L20.1154064,5.7720217 L23.0044543,8.66285128 L8.65928785,23 L5.76370704,20.1100613 L6.26021111,19.620981 C6.51321437,19.3759955 6.64951543,19.0359852 6.64981238,18.6796426 C6.64981238,18.3232999 6.51172961,17.9889318 6.26110196,17.7380072 C5.74202953,17.2186378 4.89720055,17.2180439 4.37753422,17.7371163 L4.37753422,17.7371163 L3.8869692,18.2261966 L1,15.3371487 L15.347542,1 Z M15.3466512,2.96018141 L12.2105391,6.09391788 L12.8454229,6.74246146 L11.7841158,7.78090327 L11.1602192,7.14334692 L2.95958751,15.3377426 L3.95140782,16.3298599 C5.00944848,15.7119024 6.36176876,15.8776017 7.24134114,16.758065 C8.12091353,17.6385282 8.285722,18.9905515 7.66717059,20.0485922 L7.66717059,20.0485922 L8.65869395,21.0410064 L16.8002322,12.9054072 L16.0617121,12.1508517 L17.1227222,11.1121129 L17.8505521,11.8559782 L21.0457576,8.66344519 L20.0539373,7.67103096 C18.9955997,8.28869152 17.6438733,8.12328915 16.764004,7.24282591 C15.8841347,6.36236266 15.7196231,5.01004238 16.3384715,3.95259563 L16.3384715,3.95259563 L15.3466512,2.96018141 Z M14.2708231,8.19889318 L15.6967875,9.65544367 L14.6357773,10.6941824 L13.2101099,9.23763194 L14.2708231,8.19889318 Z"
                  />
                </g>
              </svg>
            </div>
            <div
              className="iconTextWrapper"
            >
              Ilmainen
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
